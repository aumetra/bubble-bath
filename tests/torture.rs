use bubble_bath::BubbleBath;
use insta::assert_snapshot;
use std::fs;

#[test]
fn torture() {
    insta::glob!("inputs/*", |path| {
        let input = fs::read_to_string(path).unwrap();
        assert_snapshot!(bubble_bath::clean(&input).unwrap());
    });
}

#[test]
fn torture_escaped() {
    insta::glob!("inputs/*", |path| {
        let input = fs::read_to_string(path).unwrap();
        let bubble_bath = BubbleBath {
            preserve_escaped: true,
            ..BubbleBath::default()
        };

        assert_snapshot!(bubble_bath.clean(&input).unwrap());
    });
}
