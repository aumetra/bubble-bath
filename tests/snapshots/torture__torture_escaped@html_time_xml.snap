---
source: tests/torture.rs
expression: bubble_bath.clean(&input).unwrap()
input_file: tests/inputs/html_time_xml
---
&lt;html&gt;&lt;body&gt;
<?xml:namespace prefix="t" ns="urn:schemas-microsoft-com:time">
<?import namespace="t" implementation="#default#time2">
&lt;t:set attributename="innerHTML" to="XSS&lt;SCRIPT DEFER&gt;alert(" xss")&lt;="" script=""&gt;&amp;quot;&amp;gt;
&lt;/body&gt;&lt;/html&gt;
